### ISSUE TITLE
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js): Failed to parse source map

### ISSUE THREAD

**Issue by andisaba** [morellodev/react-awesome-reveal#137]
Have used 'React Awesome Reveal' with React 17... no problems... works extremely well!

However, I have installed and tried to implement in current React 18 project and now receiving 100 errors in terminal... basically all reveal error types ...the following is an example:

> **WARNING** in ./node_modules/react-awesome-reveal/dist/module.js
> **Module Warning** (from ./node_modules/source-map-loader/dist/cjs.js):
> **Failed** to parse source map from '.../node_modules/react-awesome-reveal/dist/fadeIn.ts' file: **Error:** ENOENT: no such file or directory, open '.../node_modules/react-awesome-reveal/dist/fadeIn.ts'

Is 'React Awesome Reveal' not compatible with react 18 yet OR is there some other possible explanation that I am overlooking?

Thanks in advance for your prompt reply!

Andrew

**Comment by morellodev** [comment]
React Awesome Reveal is fully compatible with React 18 starting from v4, this is not an issue related to React 18 support.

It seems your setup tries to read the source maps from an invalid path (source maps are shipped with the package), may you give me additional details?

**Comment by andisaba** [comment]
OK ... my only main difference between my last project and this one is Material UI installed.
Dependancies include:
    
    "@mui/icons-material": "^5.8.4",
    "@mui/material": "^5.9.0",
    "@mui/styled-engine-sc": "^5.8.0",
    "react-awesome-reveal": "^4.0.1"

I installed 'react-awesome-reveal' via NPM ...npm install react-awesome-reveal

**No** errors on install shown!

In my component:

> import { Fade } from 'react-awesome-reveal;

Wrapped 2 divs as shown(these are styled divs) :
     

>       <Fade>
>         <HeaderTitleText>Hello</HeaderTitleText>
>         <HeaderTitleText1>World</HeaderTitleText1>
>       </Fade>

The problem seems to be in the import phase... so if I comment out the import there are no errors
Any suggestions?

**Comment by animesh28** [comment]
This issue persists on my end too, I've got 100 compile warnings from one import 😢

**Comment by OlegGlo** [comment]
Same issue here.

**Comment by fidaay** [comment]
I got the same issue andisaba presents.

**Comment by Soapalin** [comment]
same for you and I also use Material UI...

**Comment by fidaay** [comment]
To anyone stopping by, I did this with react-app-rewired to neutralize the warning messages:

```
module.exports = {
  webpack: function (config, env) {
    return {
      ...config,
      stats: 'none',
      ignoreWarnings: [/Failed to parse source map/],
    };
  },
};

```

**Comment by morellodev** [comment]
So I guess you are using `create-react-app`, it was not specified when the issue was opened. If I have time I will look into this, but it is not an issue with the lib itself.

**Comment by fidaay** [comment]
> Yes confirmed ...used `create-react-app` ...would be great if you could look into this and indicate a fix!
> 
> Ignoring warnings as indicated by **'fidaay'** isn't addressing the underlying issue

I shared what I did, didn't indicate anything. In that way I can develop without seeing 100 warning messages.

**Comment by morellodev** [comment]
Fixed in [v4.1.0](https://github.com/morellodev/react-awesome-reveal/releases/tag/v4.1.0)

**Comment by andisaba** [comment]
Thanks for the fix... that's fantastic!

I'm keen to give it another go... fingers crossed!

Cheers!
